<launch>
  <arg name="jn0" default="false"/>
  <arg name="control" default="true"/>
  <arg name="head" default="true"/>
  <arg name="arms" default="true"/>
  <arg name="gripper" default="true"/>
  <arg name="facial_expression" default="true"/>
  <arg name="point_detail" default="true"/>
  <arg name="scene_finder" default="true"/>
  <arg name="prod" default="true"/>

  <group if="$(arg jn0)">
    <include file="$(find jn0_gazebo)/launch/jn0_empty_world.launch" />
  </group>

  <group if="$(arg control)">
    <node name="devine_irl_control" pkg="devine_irl_control" type="node_control.py" respawn="$(arg prod)">
        <param name="is_head_activated" value="$(arg head)" />
        <param name="is_arms_activated" value="$(arg arms)" />
        <param name="is_grippers_activated" value="$(arg gripper)" />
    </node>
  </group>

  <group if="$(arg facial_expression)">
    <node name="devine_irl_control_facial_expression" pkg="devine_irl_control" type="node_facial_expression.py" respawn="$(arg prod)" />
  </group>
  
  <group if="$(arg point_detail)">
      <node name="devine_irl_control_object" pkg="devine_irl_control" type="node_object.py" respawn="$(arg prod)" />
      <node name="devine_irl_control_error" pkg="devine_irl_control" type="node_error.py" respawn="$(arg prod)" />
  </group>

</launch>
